src/js/integrations/debugsymbolicator.ts

```diff
@@ -20,255 +19,163 @@ export type ReactNativeError = Error & {
 };
 
 /** Tries to symbolicate the JS stack trace on the device. */
-export class DebugSymbolicator implements Integration {
-  /**
-   * @inheritDoc
-   */
-  public static id: string = 'DebugSymbolicator';
-  /**
-   * @inheritDoc
-   */
-  public name: string = DebugSymbolicator.id;
-
-  /**
-   * @inheritDoc
-   */
-  public setupOnce(addGlobalEventProcessor: (callback: EventProcessor) => void, getCurrentHub: () => Hub): void {
-    addGlobalEventProcessor(async (event: Event, hint: EventHint) => {
-      const self = getCurrentHub().getIntegration(DebugSymbolicator);
-
-      if (!self) {
-        return event;
-      }
-
-      if (event.exception && isErrorLike(hint.originalException)) {
-        // originalException is ErrorLike object
-        const symbolicatedFrames = await this._symbolicate(
-          hint.originalException.stack,
-          getFramesToPop(hint.originalException as Error),
-        );
-        symbolicatedFrames && this._replaceExceptionFramesInEvent(event, symbolicatedFrames);
-      } else if (hint.syntheticException && isErrorLike(hint.syntheticException)) {
-        // syntheticException is Error object
-        const symbolicatedFrames = await this._symbolicate(
-          hint.syntheticException.stack,
-          getFramesToPop(hint.syntheticException),
-        );
+export const debugSymbolicatorIntegration: IntegrationFn = () => {
```

Comment:
we should backport the integrations to a v5 release so that users can migrate to the functional integrations more incrementally.

You can look at https://github.com/getsentry/sentry-javascript/blob/fdcd1adfe65becd9c605412cdd474eb676714bc4/packages/integrations/src/dedupe.ts#L9-L45 as an example of this in `v7`.

Reply:
Yes, I agree, I'll prepare it.